p4-16/psa/PSA.mdk

```diff
@@ -608,3 +608,209 @@ matches the in parameter of the subsequent block.
 ```
 [INCLUDE=psa.p4:Programmable_blocks]
 ```
+
+
+# PSA Portability Requirements
+
+This section lists minimum requirements for a P4 implementation to
+be PSA compliant.
+
+A PSA compliant implementation is allowed to support P4 programs that
+exceed one or more of these limits.  It is also allowed to reject such
+a P4 program as 'too complex'.
+
+A PSA compliant implementation must not reject a P4 program that obeys
+all of these limitations.  Thus P4 programs that obey all of these
+limitations should be portable across all PSA compliant
+implementations.
+
+Minimum requirements for packet parsing:
+
+- 32 bits in a parser `select` expression.
+- 50 parser states.
+- 20 parser state transitions per packet.
+- 128 bytes extracted per packet.
+- 16 bytes of metadata assigned in the parser.
```

Comment:
I suppose there are parsers that pick one of several IPv6 headers from which to assign the SA and DA to metadata variables, plus one of several L4 headers from which to assign source and dest port numbers?  That would total 16+16+2+2 = 36 bytes, and we could round up to 40 bytes for a bit more headroom above that.

But feel free to suggest a number other than 40 bytes if you'd be more comfortable proposing that as a PSA minimum.